gems/aws-sdk-core/lib/aws-sdk-core/param_validator.rb

```diff
@@ -130,6 +141,7 @@ def correct_type?(ref, value, errors, context)
       case value
       when Hash then true
       when ref.shape.struct_class then true
+      when Enumerator then ref.eventstream
```

Comment:
We should discuss this a bit further. This might be rather broad in what it accepts.

Reply:
Nice catch, I'll definitely consider more careful check for this one